### ISSUE TITLE
Selecting Post Only option still sends email to some users

### ISSUE THREAD

**Issue by ccwalburn** [Automattic/jetpack#35998]
### Impacted plugin

Jetpack

### Quick summary

<img width="680" alt="Markup 2024-02-27 at 13 38 55" src="https://github.com/Automattic/jetpack/assets/426388/95bfeebc-c771-43df-ba25-fc261cc5049e">

I have a report from a WordPress.com customer where the Post Only option was selected in the Newsletter section of the Jetpack tab for a new post that was published, but two subscribers still got the post email notification. 
The site where the post was posted is on the WordPress.com Explorer plan.

The two subscribers who got the email were **not** WordPress.com users. 
The other subscribers who did not get the email were WordPress.com users. 
Is this feature supposed to suppress email notifications to all users, or just WordPress.com users?

### Steps to reproduce

I did not have time to try to reproduce but I wanted to go ahead and report in case others are experiencing the same thing.

### A clear and concise description of what you expected to happen.

I expected none of the subscribers to get the post notification email when the Post Only option was selected.

### What actually happened

Two subscribers who were not WordPress.com users got the email notification.

### Impact

One

### Available workarounds?

No but the platform is still usable

### Logs or notes

7803673-zd-a8c

**Comment by jeherve** [comment]
I cannot seem to reproduce on my end. It may be good for the site owner to give it another try with a new post, if that's possible.

**Comment by ccwalburn** [comment]
Thanks for testing this! I asked the site owner if they could try again with a new post to see if it is still happening to them.

**Comment by masperber** [comment]
@jeherve I have been able to reproduce this issue on my test sites, after a customer reported it to me.

Here are the testing requirements. I am not sure if all are required, but if you meet all of them you should be able to reproduce:

* Simple site
* Site is connected to Facebook using the Jetpack Publicize feature
* The test post has a title, image, and text
* The test post has a category selected at the time of publishing

**Comment by masperber** [comment]
8062190-zen Customer report, requests followup when resolved.

**Comment by jeherve** [comment]
> if you meet all of them you should be able to reproduce:

I've had no luck reproducing, with those exact requirements. :(

@automattic/zap Do you have any idea on this?

**Comment by simison** [comment]
Thread p1714993125293119-slack-C052XEUUBL4

Code that might help D147226-code

> TLDR, some things are hooked to `post_published` and that's being fired before saving the meta for api requests
